libs/libcommon/src/libcommon/queue.py

```diff
@@ -600,6 +609,47 @@ def get_dataset_pending_jobs_for_type(self, dataset: str, job_type: str) -> List
             )
         ]
 
+    def heartbeat(self, job_id: str) -> None:
+        job = Job.objects.with_id(job_id)  # type: ignore
+        if job is not None and isinstance(job, Job):
+            job.update(last_heartbeat=get_datetime())
+
+    def get_zombies(self, max_seconds_without_heartbeat: int) -> List[JobInfo]:
+        started_jobs = Job.objects(status=Status.STARTED)
+        if max_seconds_without_heartbeat > 0:
```

Comment:
nit: I generally return as soon as possible when a condition is not met, ie start the method with:

```
if max_seconds_without_heartbeat <= 0:
  return []
started_jobs = Job.objects(status=Status.STARTED)
zombies = [
  ...
```

It de-indents a bit and helps readability, I think